concept/type/impl/ThingTypeImpl.java

```diff
@@ -279,46 +296,49 @@ private void ownsAttribute(AttributeTypeImpl attributeType, AttributeTypeImpl ov
     }
 
     @Override
-    public FunctionalIterator<AttributeTypeImpl> getOwns() {
+    public Seekable<AttributeType, Order.Asc> getOwns() {
         return getOwns(false);
     }
 
     @Override
-    public FunctionalIterator<AttributeTypeImpl> getOwnsExplicit() {
+    public Seekable<AttributeType, Order.Asc> getOwnsExplicit() {
         return getOwnsExplicit(false);
     }
 
     @Override
-    public FunctionalIterator<AttributeTypeImpl> getOwns(AttributeType.ValueType valueType) {
+    public Seekable<AttributeType, Order.Asc> getOwns(AttributeType.ValueType valueType) {
         return getOwns(valueType, false);
     }
 
     @Override
-    public FunctionalIterator<AttributeTypeImpl> getOwnsExplicit(AttributeType.ValueType valueType) {
+    public Seekable<AttributeType, Order.Asc> getOwnsExplicit(AttributeType.ValueType valueType) {
         return getOwnsExplicit(valueType, false);
     }
 
     @Override
-    public FunctionalIterator<AttributeTypeImpl> getOwns(boolean onlyKey) {
-        if (isRoot()) return Iterators.empty();
-        Set<AttributeTypeImpl> overridden = overriddenOwns(onlyKey, false).toSet();
-        assert getSupertype() != null;
-        return link(declaredOwns(onlyKey), getSupertype().getOwns(onlyKey).filter(key -> !overridden.contains(key)));
```

Comment:
TODO this behaviour is not exactly replicated in the new change! We should only filter the supertypes' owns as we do in the old code